app/src/main/java/com/andrew/tasky/presentation/agenda/AgendaViewModel.kt

```diff
@@ -72,43 +58,39 @@ class AgendaViewModel@Inject constructor(
                 add(
                     indexOfTimeNeedle(
                         agendaItems = items,
-                        currentDateTime = currentDateAndTime.value
+                        currentDateTime = currentTime
                     ),
                     UiAgendaItem.TimeNeedle
                 )
             }
     }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())
 
+    fun switchIsDone(agendaItem: AgendaItem) {
+        viewModelScope.launch {
+            val updatedAgendaItem = agendaItem.copy(isDone = !agendaItem.isDone)
+            repository.upsert(updatedAgendaItem)
+        }
+    }
+
     fun setDateSelected(dateUserSelected: LocalDate) {
         _dateSelected.value = dateUserSelected
-        _calendarDateItemList.value = calendarDateItemList.value.map { item ->
+    }
+
+    private val daysAfterCurrentDate = 5
+    val calendarDateItemList = combine(
+        dateSelected, currentDateAndTimeFlow
+    ) { dateSelected, currentDateTime ->
+        (0..daysAfterCurrentDate).map { days ->
             CalendarDateItem(
-                isSelected = dateUserSelected == item.date,
-                date = item.date
+                isSelected = dateSelected == currentDateTime.toLocalDate().plusDays(days.toLong()),
```

Comment:
Maybe I'm getting this wrong, but why do you call `plusDays(days.toLong())` here?
Shouldn't the selected date simply be the currentDate?
`isSelected = dateSelected == currentDateTime.toLocalDate()`

Reply:
If I use only currentDateTime.toLocalDate() it only compares the date I select on the mini calendar to the date it currently is (LocalDate.now()). So if I removed plusDays(days.toLong()) it'll highlight all days if the dateSelected is today's date and not highlight any if the dateSelected is any other day